test/modules/TestResultCoordinator/Services/TestResultReportingService.cs

```diff
@@ -22,25 +22,39 @@ class TestResultReportingService : IHostedService, IDisposable
         readonly TimeSpan delayBeforeWork;
         readonly ITestOperationResultStorage storage;
         readonly TestResultReportingServiceSettings serviceSpecificSettings;
+        readonly TimeSpan sendReportFrequency;
         Timer timer;
 
         public TestResultReportingService(ITestOperationResultStorage storage)
         {
             this.storage = Preconditions.CheckNotNull(storage, nameof(storage));
-            this.delayBeforeWork = Settings.Current.TestStartDelay + Settings.Current.TestDuration + Settings.Current.DurationBeforeVerification;
+            if (TestMode.Connectivity.Equals(Settings.Current.TestMode))
```

Comment:
nit: consider if new switch expression syntax would make it more compact/readable? https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression